src/pocketmine/network/mcpe/protocol/StartGamePacket.php

```diff
@@ -130,6 +134,8 @@ class StartGamePacket extends DataPacket{
 	/** @var bool */
 	public $onlySpawnV1Villagers = false;
 
+	/** @var string */
+	public $vanillaVersion = ProtocolInfo::MINECRAFT_VERSION_NETWORK;
```

Comment:
I think it’s better to leave an empty string here - the buffer will be a couple of bytes less. This field is useless.

Also `vanillaVersion` refers to LevelSettings, you put newline in the wrong place.

Reply:
From what I've heard, sending anything other than the version will crash the client.

Reply:
Personally, I send an empty string there and everything works fine...

Reply:
works for you != works for everyone

haggling over a premature optimisation like that is a waste of everyone's time.

Reply:
> works for you != works for everyone

But what if I say that everyone who uses my fork with 1.13 support works fine?
By the way, I want to note that BDS itself sends an empty string there.

Reply:
It doesn't matter either way. This is just bikeshedding over a micro optimisation. Let it go.